doc/specs/#653 - Quake Mode/#653 - Quake Mode.md

```diff
@@ -0,0 +1,629 @@
+* The window to no longer appear on the taskbar
+* The window to no longer appear in the alt-tab order
+
+When minimized to the tray, it's almost as if there's no window for the Terminal
+at all. This can be combined with the global hotkey to quickly restore the window.
+
+The tray icon could be used for a variety of purposes. As a simple start, we
+could include the following three options:
+
+```
+Focus Terminal
+---
+Windows > Window 1 - <un-named window>
+          Window 2 - "This-window-does-have-a-name"
+---
+Quit
+```
+
+Just clicking on the icon would summon the recent terminal window. Right
+clicking would show the menu with "Focus Terminal", "Windows" and "Quit" in it, and
+"Windows" would have nested entries for each Terminal window.
+
+* "Focus Terminal" would do just that - summon the most recent terminal window,
+  wherever it is.
+* "Windows" would have nested popups for each open Terminal window. Each of
+  these nested entries would display the name and ID of the window. Clicking
+  them would summon that window (wherever it may be)
+* "Quit" would be akin to quit in browsers - close all open windows
+  <sup>[[1]](#footnote-1)</sup>.
+
+The tray notification would be visible always when the user has
+`"minimizeToTray": true` set in their settings. If the user has that set to
+false, but would still like the tray, they can specify `"alwaysShowTrayIcon":
+true`. That will cause the tray icon to always be added to the system tray.
+
+There's not a combination of settings where the Terminal is "minimized to the
+tray", and there's _no tray icon visible_. We don't want to let users get into a
+state where the Terminal is running, but is totally hidden from their control.
+
+From a technical standpoint, the tray icon is managed similar to the global
+hotkey. The Monarch process is responsible for setting it up, and processing the
+messages. When a Monarch dies and a new process becomes the Monarch, then it
+will re-create the tray icon.
+
+## UI/UX Design
+
+To summarize, we're proposing the following set of settings:
+
+```jsonc
+{
+    "minimizeToTray": bool,
+    "alwaysShowTrayIcon": bool,
+    "globalHotkeys": [
+        {
+            "keys": KeyChord,
+            "dropdownDuration": float,
+            "hideWhenVisible": bool,
+            "monitor": "any"|"toCurrent"|"onCurrent"|int,
+            "desktop": null|"toCurrent"|"onCurrent"
+        }
+    ]
+}
+```
+
+## Potential Issues
+
+<table>
+
+<tr>
+<td><strong>Compatibility</strong></td>
+<td>
+
+As part of this set of changes, we'll also be allowing the <kbd>Win</kbd> key in
+keybindings. Generally, the OS reserves the Windows key for its own shortcuts.
+For example, <kbd>Win+R</kbd> for the run dialog, <kbd>Win+A</kbd> for the
+Action Center, <kbd>Win+V</kbd> for the cloud clipboard, etc. Users will now be
+able to use the win key themselves, but they should be aware that the OS has
+"first dibs" on any hotkeys involving the Windows key.

```

Comment:
Can we detect which ones are taken and throw a warning when that happens? Rather than silently fail.

Reply:
> If that should happen, then we should display a
> warning dialog to the user indicating which hotkey will not work (because it's
> already used for something else).